src/utils/error.middleware.ts

```diff
@@ -1,15 +1,40 @@
-/* eslint-disable max-len */
 /* eslint-disable no-unused-vars */
 import { AxiosError } from 'axios';
+import * as apm from 'elastic-apm-node';
 import * as express from 'express';
 
-export default (error: any, _req: express.Request, res: express.Response, next: express.NextFunction) => {
+/**
+ * Express Error Handler.
+ * The handler catch the error thrown by a request and send it to the user.
+ * If the error is an axios error, details of the error will be sent to the client.
+ * Also, if there is an open transaction it will be closed as an error.
+ */
+export default (error: any, _req: express.Request,
+  res: express.Response, next: express.NextFunction) => {
+  const transaction = apm.currentTransaction;
+
   if (error?.isAxiosError) {
     const err = error as AxiosError;
     const status = err.response?.status;
     const data = err.response?.data;
+
+    // send error to apm
+    apm.captureError(err);
+    if (transaction) {
+      transaction.result = 'error';
+      transaction.end();
+    }
```

Comment:
this code is done anyway. so maybe write it before the if statement

Reply:
@lustigy96  I'm not really sure what you mean by:

> this code is done

Can you please explain?